src/Rings/MPolyMap/MPolyQuo.jl

```diff
@@ -86,7 +86,11 @@ end
 # The two main evaluation methods
 function (F::MPolyAnyMap{<: MPolyQuo})(g)
   if g isa elem_type(domain(F))
-    return _evaluate_plain(F, g)
+    if coefficient_map(F) === nothing
```

Comment:
Why don't you resolve this via dispatch as in the fix I proposed? Then the `if`-clause becomes superfluous. Am I missing something?

Reply:
Because then it works only if the coefficient map is an `MPolyAnyMap` itself. But there might be many other maps. I don't want to do dispatch, because then I have to duplicate more code.

Reply:
I don't see why. Couldn't you just make a dispatch for the case where the type `T` of the coefficient map is any instance of the `AbstractAlgebra.Map` rather than `Nothing`? That wouldn't be as restrictive. And the code duplication that I see at this point is not too much more than what comes from the `if-else` block. Or does this entail some more difficulties further down the road?

Reply:
No I don't think this works. We also allow arbitrary julia functions as coefficient maps, e.g., `hom(A, A, a -> a^2, [x, y])` which can be very handy.

As this does not affect functionality or anything else, I don't want to duplicate more code at this point.

Reply:
Fair enough. Allowing julia functions to map coefficients is a good argument.